.github/workflows/testing.yml

```diff
@@ -32,10 +32,17 @@ jobs:
           git config --global user.email "CI@example.com"
           git config --global user.name "CI Almighty"
           pip install wheel # Needed for scan image
-      - name: Install nwb-conversion-tools with minimal requirements
+          
+      - name: Install with minimal requirements
         run: pip install .[test]
       - name: Run minimal tests
-        run: pytest tests/test_internals -n auto --dist loadscope
+        run: pytest tests/test_minimal -n auto --dist loadscope
+        
+      - name: Install with ophys requirements
+        run: pip install .[ophys]
+      - name: Run ophys tests
+        run: pytest tests/test_ophys -n auto --dist loadscope
```

Comment:
This demonstrates one possible way we could deploy tests for the specific keys. Personally I think I'd be fine with just one of these steps per each modality.

Another option would be to utilize a matrix, which would use a fresh environment for each fresh hyper-specific installation. This would be the strongest form of testing for this as it would truly represent the intended user workflow. However, it would result in an awful lot of action deployments, which could get spammy and could be considered a fair amount of 'wasted' computational resources if set to trigger on every push. I'd say if that's the route we want to go it would be better to have it set as a daily schedule (or longer).

Reply:
I think we should deploy the tests somewhere where the basic version has not be installed before as this is the way that I think the functionality will be used. That is, it has more ecological validity. What do you think?

Reply:
Both ways need to work - I'm finding it very convenient personally to start from a minimal base (probably the first thing people will intuitively think to do) then if I need to add something (or more than one thing) to use an interface I can just do `install neuroconv[format_name]` and it will quickly skip the installation of everything except for the items needed for what I requested

Reply:
Although, don't you think `setuptools` are trustworthy enough to know they will also install minimal requirements any time an extra key is specified? I don't think that's something we would actually have to explicitly test.

I can add a separate gallery for each individual format key (starting from a minimal base) though, to ensure they did in fact wind up in the `extra_requires` with the correct dependencies to run

Reply:
> I'm finding it very convenient personally to start from a minimal base (probably the first thing people will intuitively think to do) then if I need to add something (or more than one thing) to use an interface I can just do install neuroconv[format_name] and it will quickly skip the installation of everything except for the items needed for what I requested

So I guess this is the key point. My impression is that it would be more common for users to just install pip install neuroconv[modality] but it could be that they first install the minimum and they go for the modalities as you describe. I don't know what is more common. 

As far as I remember, pip install package[modality] runs both the minimal+extra_requirements at the same time. My feeling is that dependency resolution will work differently in these two cases (sequential with minimal then modality vs installing single modality) so different dependencies might end up being installed on each case.

Reply:
> As far as I remember, pip install package[modality] runs both the minimal+extra_requirements at the same time.

Yep!

> My feeling is that dependency resolution will work differently in these two cases (sequential with minimal then modality vs installing single modality) so different dependencies might end up being installed on each case.

Hence the setup for https://github.com/catalystneuro/neuroconv/pull/115 to test that at least to first-order. Should be 'good enough'?